recipes/libcal/meta.yaml

```diff
@@ -0,0 +1,114 @@
+{% set version = "0.4.3.36" %}
+{% set lib_version = version.split(".")[0]+"."+version.split(".")[1]+"."+version.split(".")[2]+".0" %}
+{% set platform = "linux-x86_64" %}  # [linux64]
+{% set platform = "linux-sbsa" %}    # [aarch64]
+{% set extension = "tar.xz" %}
+
+package:
+  name: libcal
+  version: {{ version }}
+
+source:
+  url: https://developer.download.nvidia.com/compute/cublasmp/redist/libcal/{{ platform }}/libcal-{{ platform }}-{{ version }}_cuda{{ cuda_version }}-archive.{{ extension }}
+  sha256: 271d9951ee5b5e4e6e9fcb7079ce3c36f5dd581b3a0f7d1937f81e0c0e058fb8  # [linux64 and cuda_version == 11]
+  sha256: eac07cc1524b37c693944e3e91c4857ed66bc2bb2579407ce9795916c9a113a4  # [linux64 and cuda_version == 12]
+  sha256: d0016b62d42916e37cd065279eb8fa4947e3423220812c696c09ea417477c770  # [aarch64 and cuda_version == 11]
+  sha256: 3500c18236fc449cf0395ee422c5a1771ce6349dc547951eaa6562ffc4fc762c  # [aarch64 and cuda_version == 12]
+
+build:
+  number: 0
+  skip: true  # [not (linux64 or aarch64)]
+  missing_dso_whitelist:
+    - "*/libcuda.so.*"
+  script:
+    - cp -rv include $PREFIX
+    - cp -rv lib $PREFIX
+    - cp -rv share $PREFIX
+    - cp -rv src $PREFIX
```

Comment:
Q: What's in the `src` dir?

Reply:
(for my understanding) This does not affect the output conda package, does it? Though same question applies to libcal-dev where output includes the src/.

I'll clarify with cublasmp developers what is src/  directory and ucc.conf file for exactly.

Reply:
In a conda recipe, the `build:script:` section refers to how this conda package is going to be built. The `cp` commands here are saying: Let us move the extracted files from the current dir (`$SRC_DIR`) to the package location (`$PREFIX` pointing to the "host" environment). So, no, it does not "affect" the package (in the negative sense).

Reply:
`src/cal-mpi/cal_mpi.c` and `src/cal-mpi/cal_mpi.h` contain helper functions that can be used by users to bootstrap CAL (and UCC) communicator. It is my understanding that they are provided there for convenience, and they are repeated in the documentation (https://docs.nvidia.com/cuda/cusolvermp/usage/cal.html#creating-communicator-handle-with-mpi).

Reply:
`share/ucc.conf` is the default UCC configuration file used by libcal, and users can point to a custom config using the `UCC_CONFIG_FILE ` environment variable (https://docs.nvidia.com/cuda/cusolvermp/usage/cal.html#configuring-communications-submodule).

Reply:
@mrogowski Any chance you know UCC itself or any other layers would also automatically pick up `share/ucc.conf`? If so, we might need to move it to a subdirectory under `share`.

Reply:
@leofang it looks like UCC will look for it under `<ucc_install_dir>/share/ucc.conf` (https://github.com/openucx/ucc/blob/d82f0f62d55524853d1ab5a6a4b23f7ae06181f3/src/core/ucc_constructor.c#L59)

Reply:
I see, good finding Marcin! We should place it in a subdir and ask users to opt-in.

Reply:
libcal will currently look for `ucc.conf` in `/usr/share`, `/usr/share/libcal-cuda-11`, `/usr/share/libcal-cuda-12`, then `<library_path>/../share/`. I think only the first location can be found by UCC, and only if it's installed in `/`. Perhaps better to use one of those so that libcal still finds the default.

Reply:
@mrogowski The purpose of the renaming is to _not_ allow it to be picked up automatically. We need to document it so that users know where to find the file. Since conda-forge support is brand-new to cuBLASMp, you will need to update the documentation to mention conda-forge anyway.

If we have a file `$PREFIX/share/ucc.conf` then UCC would automatically pick it up based on [this logic](https://github.com/openucx/ucc/blob/4b1ebbc1083fdba5a8cfa25310e5e70bc0cad9fe/src/core/ucc_constructor.c#L60), even if users are not running cuBLASMp and just have it installed in the current conda environment for whatever reason. On conda-forge, we work very hard to avoid interfering with other packages.

Reply:
If we want, after the feedstock is created we can add a post-link script like what UCX did to give users performance instructions:
https://github.com/conda-forge/ucx-split-feedstock/blob/main/recipe/ucx-post-link.sh

Reply:
I think we want libcal to pick it up automatically, but not UCC - hence my comment on where libcal will search for it.

Reply:
Then libcal needs to change because I was saying `<library_path>/../share/ucc.conf` will also be picked up by UCC.

Reply:
My understanding is that the files installed to `$PREFIX/src` are optional, and you don't want users to get them by default. In that case, the recipe should be modified so they are separated into an additional recipe output that can be installed optionally. e.g. `libcal-dev-mpi` or something.

Reply:
Small update on this: we can get rid of `share/`, `share/ucc.conf`, and `src/`.

The conda package for UCC doesn't contain a `ucc.conf`, and other UCC distributions seem to have changed the variables they export. This file was meant to overwrite what their config was setting for libcal. Future versions of libcal will not include `ucc.conf` either. As @carterbox said, the files in `src/` are optional. We don't necessarily need them in this context. The same content is in the documentation.

CC @almogsegal